### PR TITLE
Fix several scrutinizer issues

### PR DESCRIPTION
**Pull request by Ener-Getick**
This PR fix several scrutinizer issues.

### CODE REVIEW

File: Controller/ExceptionController.php [hunk 1]
```diff
@@ -126,7 +126,7 @@ protected function getAndCleanOutputBuffering()
      * @param FlattenException $exception
      * @param array            $exceptionMap
      *
-     * @return int|false
+     * @return mixed False, or the $exceptionMap corresponding value
```
Comment (stof) [inline Controller/ExceptionController.php]: this is wrong. The ``$exceptionMap`` array is meant to contain only integers as values

File: Request/ParamFetcher.php [hunk 2]
```diff
@@ -95,6 +96,8 @@ public function getParams()
 
     /**
      * {@inheritdoc}
+     *
+     * @throws \InvalidArgumentException
```
Comment (stof) [inline Request/ParamFetcher.php]: I don't think this is needed here. the case where it happens is not a case where you would want to catch it (except on a top-level error handler)

File: Controller/ExceptionController.php [hunk 3]
```diff
@@ -141,7 +141,7 @@ protected function isSubclassOf($exception, $exceptionMap)
                 }
             }
         } catch (\ReflectionException $re) {
-            return 'FOSUserBundle: Invalid class in  fos_res.exception.messages: '
+            return 'FOSUserBundle: Invalid class in fos_res.exception.messages: '
```
Comment (Ener-Getick) [inline Controller/ExceptionController.php]: Why the exception is not thrown here ?

File: Request/ParamFetcher.php [hunk 4]
```diff
@@ -188,6 +191,21 @@ public function all($strict = null)
...
     /**
+     * @throws \InvalidArgumentException
+     *
+     * @return Request
+     */
+    private function getRequest()
+    {
+        $request = $this->requestStack->getCurrentRequest();
+        if ($request === null) {
+            throw new \InvalidArgumentException('There is no current request.');
```
Comment (stof) [inline Request/ParamFetcher.php]: a RuntimeException would be better. This is not about validating an argument

File: Request/ParamFetcher.php [hunk 5]
```diff
@@ -188,6 +191,21 @@ public function all($strict = null)
     }
 
     /**
+     * @throws \InvalidArgumentException
+     *
+     * @return Request
+     */
+    private function getRequest()
```
Comment (stof) [inline Request/ParamFetcher.php]: private methods should be after protected ones

File: Controller/ExceptionController.php (hunk 1 above)
Comment (Ener-Getick) [inline Controller/ExceptionController.php]: No, the ``$exceptionMap`` can also contain boolean.
https://github.com/FriendsOfSymfony/FOSRestBundle/blob/master/DependencyInjection/Configuration.php#L327
https://github.com/FriendsOfSymfony/FOSRestBundle/blob/master/Controller/ExceptionController.php#L161

File: Request/ParamFetcher.php (hunk 5 above)
Comment (Ener-Getick) [inline Request/ParamFetcher.php]: Yes, we should probably fix all this mistakes in another PR.

File: Controller/ExceptionController.php (hunk 1 above)
Comment (Ener-Getick) [inline Controller/ExceptionController.php]: I replace ``false`` by ``bool`` but the class can send a string, do you know why ?

File: Request/ParamFetcher.php (hunk 4 above)
Comment (Ener-Getick) [inline Request/ParamFetcher.php]: You're right, I changed it.

File: Controller/ExceptionController.php (hunk 3 above)
Comment (lsmith77) [inline Controller/ExceptionController.php]: since we are inside the exception handler, we would otherwise risk a loop.

Comment (xabbuh) [inline Controller/ExceptionController.php]: Then we need to add `string` as a possible return type too.

**Pull request merged**